provider/equinix/instance.go

```diff
@@ -30,11 +33,53 @@ func (device *equinixDevice) Id() instance.Id {
 }
 
 func (device *equinixDevice) Status(ctx context.ProviderCallContext) instance.Status {
-	panic(errors.NewNotImplemented(nil, "not implemented"))
+	var jujuStatus status.Status
+
+	switch device.State {
+	case "provisioning":
+		jujuStatus = status.Pending
+	case "active":
+		jujuStatus = status.Running
+	case "shutting-down", "terminated", "stopping", "stopped":
+		jujuStatus = status.Empty
+	default:
+		jujuStatus = status.Empty
+	}
+
+	return instance.Status{
+		Status:  jujuStatus,
+		Message: device.State,
+	}
+
 }
 
 // Addresses implements network.Addresses() returning generic address
 // details for the instance, and requerying the ec2 api if required.
 func (device *equinixDevice) Addresses(ctx context.ProviderCallContext) (corenetwork.ProviderAddresses, error) {
-	panic(errors.NewNotImplemented(nil, "not implemented"))
+	var addresses []corenetwork.ProviderAddress
+
+	for _, netw := range device.Network {
+		address := corenetwork.ProviderAddress{}
+		address.Value = netw.Address
+		address.CIDR = fmt.Sprintf("%s/%d", netw.Network, netw.CIDR)
+
+		if netw.Public {
+			address.Scope = corenetwork.ScopePublic
+		} else {
+			address.Scope = corenetwork.ScopeCloudLocal
+		}
+
+		if netw.AddressFamily == 4 {
+			address.Type = network.IPv4Address
+		} else {
+			address.Type = network.IPv6Address
+			logger.Infof("skipping IPv6 Address %s", netw.Address)
+
+			continue
```

Comment:
We should run some tests to check if we actually need to skip this. Juju should be able to work with IPV6 addresses.